crates/binjs_io/src/io/content.rs

```diff
@@ -0,0 +1,120 @@
+    {
+        ContentInfo {
+            bools: f("bools"),
+            floats: f("floats"),
+			unsigned_longs: f("unsigned_longs"),
+			string_enums: f("string_enums"),
+			property_keys: f("property_keys"),
+			identifier_names: f("identifier_names"),
+			interface_names: f("interface_names"),
+			string_literals: f("string_literals"),
+			list_lengths: f("list_lengths"),
+        }
+    }
+
+    /// Convert a `ContentInfo` into another one.
+    pub fn into_with<F, U>(self, f: F) -> ContentInfo<U>
+        where F: Fn(T, &str) -> U
+    {
+        ContentInfo {
+			bools: f(self.bools, "bools"),
+			floats: f(self.floats, "floats"),
+			unsigned_longs: f(self.unsigned_longs, "unsigned_longs"),
+			string_enums: f(self.string_enums, "string_enums"),
+			property_keys: f(self.property_keys, "property_keys"),
+			identifier_names: f(self.identifier_names, "identifier_names"),
+			interface_names: f(self.interface_names, "interface_names"),
+			string_literals: f(self.string_literals, "string_literals"),
+			list_lengths: f(self.list_lengths, "list_lengths"),
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (&T, &'static str)> {
+        vec![
+            (&self.bools, "bools"),
+            (&self.floats, "floats"),
+			(&self.unsigned_longs, "unsigned_longs"),
+			(&self.string_enums, "string_enums"),
+			(&self.property_keys, "property_keys"),
+			(&self.identifier_names, "identifier_names"),
+			(&self.interface_names, "interface_names"),
+			(&self.string_literals, "string_literals"),
+			(&self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = (&mut T, &'static str)> {
+        vec![
+            (&mut self.bools, "bools"),
+            (&mut self.floats, "floats"),
+			(&mut self.unsigned_longs, "unsigned_longs"),
+			(&mut self.string_enums, "string_enums"),
+			(&mut self.property_keys, "property_keys"),
+			(&mut self.identifier_names, "identifier_names"),
+			(&mut self.interface_names, "interface_names"),
+			(&mut self.string_literals, "string_literals"),
+			(&mut self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+
+    pub fn into_iter(self) -> impl Iterator<Item = (T, &'static str)> {
+        vec![
+            (self.bools, "bools"),
+            (self.floats, "floats"),
+			(self.unsigned_longs, "unsigned_longs"),
+			(self.string_enums, "string_enums"),
+			(self.property_keys, "property_keys"),
+			(self.identifier_names, "identifier_names"),
+			(self.interface_names, "interface_names"),
+			(self.string_literals, "string_literals"),
+			(self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+}
+
+impl<T> std::iter::FromIterator<(T, &'static str)> for ContentInfo<Option<T>> {
+    fn from_iter<U>(iter: U) -> Self
+    where
+        U: IntoIterator<Item = (T, &'static str)>
+    {
+        let mut container = Self::with(|_| None);
+        for (value, name) in iter.into_iter() {
```

Comment:
can you tell me what this block actually does?
this mostly looks nop+assertion but not sure what area/scope `replace` affects

Reply:
We replace the content of the field (`container.bools`, `container.floats`, etc.) with the `value` extracted from the iterator.

`replace(...)` returns the old value, which lets us assert that we haven't inserted two values for the same field

Reply:
I still don't understand the semantics (or maybe higher overview of what this method i for)
what does this method actually do? why do you replace?